homeassistant/components/mold_indicator/config_flow.py

```diff
@@ -0,0 +1,100 @@
+"""Config flow for Mold Indicator integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_NAME
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import (
+    CONF_CALIBRATION_FACTOR,
+    CONF_INDOOR_HUMIDITY,
+    CONF_INDOOR_TEMP,
+    CONF_OUTDOOR_TEMP,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ISSUE_PLACEHOLDER = {"url": "/config/integrations/dashboard/add?domain=mold_indicator"}
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_INDOOR_TEMP): selector.EntitySelector(
```

Comment:
I'm not sure whether this is possible, but ideally we would filter the entities based on device class, so the user can only select a temperature sensor instead of any sensor.

It looks like `EntitySelectorConfig` has a `filter` attribute that might allow this, but I don't see it being used anywhere so not sure if it would work.

https://github.com/home-assistant/core/blob/80a616d23761b860bd6d030c6f9756f36a2f500a/homeassistant/helpers/selector.py#L748

Reply:
Yeah, that's what I was originally trying to do but gave up as I wasn't sure how. I always ended up with it not being able to find any entities in my test environment.

Reply:
Hmm, for me that works:
```python
        vol.Required(CONF_INDOOR_TEMP): selector.EntitySelector(
            selector.EntitySelectorConfig(
                domain=SENSOR_DOMAIN,
                filter=selector.EntityFilterSelectorConfig(
                    device_class=SensorDeviceClass.TEMPERATURE
                ),
            ),
        ),
```
Then it shows only temperature sensors in the list.